tests/handlers/compression/test_xz.py

```diff
@@ -0,0 +1,704 @@
+00000010  16 00 00 00 74 2f e5 a3  e0 00 8c 00 77 5d 00 2b  |....t/......w].+|
+00000020  9a 08 27 75 19 7c b8 ad  22 99 7e ea d4 e5 70 60  |..'u.|..".~...p`|
+00000030  bf 59 c1 89 bc 13 41 40  ca 24 c6 f9 ad 99 14 34  |.Y....A@.$.....4|
+00000040  d7 d6 85 be 0c ba a8 8b  a0 d7 97 72 da 7f e7 f2  |...........r....|
+00000050  73 78 69 3d cd b5 e6 45  2f b0 23 68 22 12 c8 e4  |sxi=...E/..h"...|
+00000060  43 f2 59 1b 17 3a ad 36  08 fb 21 c9 29 ba 13 1e  |C.Y..:.6..!.)...|
+00000070  cf c3 fb 69 65 57 5a 44  80 82 4d a1 a3 fd 8e 8d  |...ieWZD..M.....|
+00000080  f0 89 5f a6 71 c9 e8 44  80 cb f1 4e d0 cc 74 45  |.._.q..D...N..tE|
+00000090  fe 86 5b 63 00 00 00 00  8c e8 0b 54 41 0a 38 d9  |..[c.......TA.8.|
+000000a0  22 81 f0 55 47 bc ab 75  e3 a0 a8 0c 70 88 35 dd  |"..UG..u....p.5.|
+000000b0  c9 e5 44 4b 5f 81 7d 25  00 01 ab 01 8d 01 00 00  |..DK_.}%........|
+000000c0  91 18 1f 93 b6 e9 df 1c  02 00 00 00 00 0a 59 5a  |..............YZ|
+"""
+)
+
+XZ_SHA256_CONTENT_BAD_IDX = unhex(
+    """\
+00000000  fd 37 7a 58 5a 00 00 0a  e1 fb 0c a1 02 00 21 01  |.7zXZ.........!.|
+00000010  16 00 00 00 74 2f e5 a3  e0 00 8c 00 77 5d 00 2b  |....t/......w].+|
+00000020  9a 08 27 75 19 7c b8 ad  22 99 7e ea d4 e5 70 60  |..'u.|..".~...p`|
+00000030  bf 59 c1 89 bc 13 41 40  ca 24 c6 f9 ad 99 14 34  |.Y....A@.$.....4|
+00000040  d7 d6 85 be 0c ba a8 8b  a0 d7 97 72 da 7f e7 f2  |...........r....|
+00000050  73 78 69 3d cd b5 e6 45  2f b0 23 68 22 12 c8 e4  |sxi=...E/..h"...|
+00000060  43 f2 59 1b 17 3a ad 36  08 fb 21 c9 29 ba 13 1e  |C.Y..:.6..!.)...|
+00000070  cf c3 fb 69 65 57 5a 44  80 82 4d a1 a3 fd 8e 8d  |...ieWZD..M.....|
+00000080  f0 89 5f a6 71 c9 e8 44  80 cb f1 4e d0 cc 74 45  |.._.q..D...N..tE|
+00000090  fe 86 5b 63 00 00 00 00  8c e8 0b 54 41 0a 38 d9  |..[c.......TA.8.|
+000000a0  22 81 f0 55 47 bc ab 75  e3 a0 a8 0c 70 88 35 dd  |"..UG..u....p.5.|
+000000b0  c9 e5 44 4b 5f 81 7d 25  01 01 ab 01 8d 01 00 00  |..DK_.}%........|
+000000c0  91 18 1f 93 b6 e9 df 1c  02 00 00 00 00 0a 59 5a  |..............YZ|
+"""
+)
+
+NULL_PADDING = b"\x00" * 4
+BAD_PADDING = b"\x00" * 3
+DATA_PADDING = b"123"
+
+
+@pytest.mark.parametrize(
+    "content, start_offset, stream_flag",
+    (
+        pytest.param(
+            XZ_NONE_CONTENT,
+            0,
+            NONE_STREAM_FLAG,
+            id="stream_flag_crc_none",
+        ),
+        pytest.param(
+            XZ_CRC32_CONTENT,
+            0,
+            CRC32_STREAM_FLAG,
+            id="stream_flag_crc_crc32",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT,
+            0,
+            CRC64_STREAM_FLAG,
+            id="stream_flag_crc_crc64",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT,
+            0,
+            SHA256_STREAM_FLAG,
+            id="stream_flag_crc_sha256",
+        ),
+    ),
+)
+def test_xz_stream_flag(content: bytes, start_offset: int, stream_flag: int):
```

Comment:
start_offset is always 0, do we need to test other values or do we need to specify it here?

maybe add negative tests as well?

Reply:
I don't know what happened, there was a mismatch between my local version and this one. Anyways, this test was removed because we no longer verify stream flags.

Reply:
Fixed together.